### PR TITLE
Bug 1401787 - Add heartbeat ping schema

### PR DESCRIPTION
**Pull request by relud**
r? @fbertsch

### CODE REVIEW

File: templates/telemetry/heartbeat/heartbeat.4.parquetmr.txt [hunk 1]
```diff
@@ -0,0 +1,46 @@
+message heartbeat {
+  required binary type (UTF8);
+  required binary id (UTF8);
+  required binary creationDate (UTF8);
+  required double version;
+  required binary clientId (UTF8);
+  required group application {
```
Comment (trink) [inline templates/telemetry/heartbeat/heartbeat.4.parquetmr.txt]: use the template https://github.com/mozilla-services/mozilla-pipeline-schemas/blob/master/templates/include/telemetry/application.1.parquetmr.txt

Comment (relud) [reply templates/telemetry/heartbeat/heartbeat.4.parquetmr.txt]: fixed

File: templates/telemetry/heartbeat/heartbeat.4.schema.json [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      "properties": {
+        "version": {
+          "maximum": 1,
+          "minimum": 1,
+          "type": "number"
+        },
+        "flowId": {
+          "type": "string"
+        },
+        "offeredTS": {
```
Comment (fbertsch) [inline templates/telemetry/heartbeat/heartbeat.4.schema.json]: Please add a tighter bound here, at least `>0` (and for the rest of the TS fields).

File: templates/telemetry/heartbeat/heartbeat.4.schema.json [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        "closedTS": {
+          "type": "integer"
+        },
+        "expiredTS": {
+          "type": "integer"
+        },
+        "windowClosedTS": {
+          "type": "integer"
+        },
+        "score": {
```
Comment (fbertsch) [inline templates/telemetry/heartbeat/heartbeat.4.schema.json]: @mythmon says this is in the range of `1-5`

File: templates/telemetry/heartbeat/heartbeat.4.schema.json [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+          "type": "integer"
+        },
+        "surveyId": {
+          "type": "string"
+        },
+        "surveyVersion": {
```
Comment (fbertsch) [inline templates/telemetry/heartbeat/heartbeat.4.schema.json]: Always a 64 chars hex

Comment (relud) [reply templates/telemetry/heartbeat/heartbeat.4.schema.json]: my bad, i was going off of http://gecko-docs.mozilla.org.s3.amazonaws.com/toolkit/components/telemetry/telemetry/data/heartbeat-ping.html but apparently this stopped being an integer over a year ago. fixed.

File: templates/telemetry/heartbeat/heartbeat.4.schema.json (hunk 3 above)
Comment (relud) [reply templates/telemetry/heartbeat/heartbeat.4.schema.json]: fixed

File: templates/telemetry/heartbeat/heartbeat.4.schema.json (hunk 2 above)
Comment (relud) [reply templates/telemetry/heartbeat/heartbeat.4.schema.json]: Added bounds of >2010 and <2100

**Comment by relud** [comment]
> Some of those fields in the payload are required, some are not; please include that as well.

I talked with @mythmon, and I think I have the required/optional fields correct now

File: templates/telemetry/heartbeat/heartbeat.4.schema.json [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+          "pattern": "^[a-fA-F0-9]{64}$",
+          "type": "string"
+        },
+        "testing": {
+          "maximum": 1,
+          "minimum": 1,
+          "type": "integer"
```
Comment (relud) [inline templates/telemetry/heartbeat/heartbeat.4.schema.json]: `testing` is apparently 1 or null according to https://github.com/mozilla/normandy/blob/master/recipe-server/client/actions/show-heartbeat/index.js#L272

**Pull request merged**